### ISSUE TITLE
different epoch when training

### ISSUE THREAD

**Issue by alicera** [ultralytics/ultralytics#909]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

last1_100.pt  and  last2_100.pt are different weights.
The difference is the epoch that training command (--epoch)
what are the different between last1_100.pt  and  last2_100.pt ( like mAP same ?
```
# get the last1_100.pt
yolo detect train data=coco128.yaml model=yolov8n.pt epochs=100 imgsz=640
```

```
# get the last2_100.pt at 100 epoch 
yolo detect train data=coco128.yaml model=yolov8n.pt epochs=300 imgsz=640
```

**Comment by glenn-jocher** [comment]
@alicera last1_100.pt and last2_100.pt are two different weight files that were created at different epochs during training. The difference between the two weights depends on how the model has learned to detect objects over time.

Typically, as the model trains for more epochs, it should be able to detect objects better and improve its mean average precision (mAP). However, it's not always the case and it's possible that one weight file performs better than another even if it was trained for fewer epochs.

To compare the mAP of the two weights, you can use the val command in YOLOv8:

```bash
yolo val data=coco128.yaml weights=last1_100.pt imgsz=640
yolo val data=coco128.yaml weights=last2_100.pt imgsz=640
```
This will output the mAP of each weight file on the COCO validation set. You can compare the mAP values to see which weight file performs better.
